app/src/main/java/io/pslab/models/PSLabSensor.java

```diff
@@ -334,6 +343,41 @@ public boolean onOptionsItemSelected(MenuItem item) {
                     isRecording = false;
                     prepareMarkers();
                 }
+            }
+            else
+            {
+                try {
+                    CustomSnackBar.showSnackBar(sensorParentView, NOT_CONNECTED, null, null, Snackbar.LENGTH_LONG);
+
+                    AlertDialog.Builder builder1 = new AlertDialog.Builder(this);
+                    builder1.setTitle(R.string.gas_sensor_not_connected);
+                    builder1.setMessage(R.string.gas_sensor_not_connected_description);
+                    builder1.setCancelable(true);
+
+                    builder1.setPositiveButton(
+                            R.string.gas_sensor_configure_button,
+                            new DialogInterface.OnClickListener() {
+                                public void onClick(DialogInterface dialog, int id) {
+                                    dialog.cancel();
+                                }
+                            });
+
+                    builder1.setNegativeButton(
+                            R.string.cancel,
+                            new DialogInterface.OnClickListener() {
+                                public void onClick(DialogInterface dialog, int id) {
+                                    dialog.cancel();
+                                }
+                            });
+                    builder1.setIcon(android.R.drawable.ic_dialog_alert);
+                    AlertDialog alert11 = builder1.create();
+                    alert11.show();
```

Comment:
No need of many dialog boxes

Reply:
ok, sure I will resolve these issues very soon. Thanks.